src/main/java/com/toptrumps/cli/CommandLineController.java

```diff
@@ -3,109 +3,215 @@
 import com.toptrumps.core.card.Attribute;
 import com.toptrumps.core.card.Card;
 import com.toptrumps.core.engine.Game;
-import com.toptrumps.core.engine.GameEventListener;
 import com.toptrumps.core.engine.RoundOutcome;
+import com.toptrumps.core.player.AIPlayer;
 import com.toptrumps.core.player.Player;
 import com.toptrumps.core.utils.ResourceLoader;
 
 import java.io.BufferedReader;
 import java.io.InputStream;
 import java.io.InputStreamReader;
+import java.util.ArrayList;
 import java.util.InputMismatchException;
 import java.util.List;
-import java.util.ArrayList;
 import java.util.Scanner;
 import java.util.stream.Collectors;
 
-public class CommandLineController implements GameEventListener {
+import static java.util.stream.Collectors.toCollection;
+
+public class CommandLineController {
 
+    private final static String DECK_RESOURCE = "assets/StarCitizenDeck.txt";
     private static final String WELCOME_BANNER_RESOURCE = "assets/banners/welcome.txt";
 
     private Scanner scanner;
     private OutputLogger logger;
 
     // TODO: Refactor the controller to use CommandLineView
     private CommandLineView view;
+    private Game gameEngine;
 
-    private static final int MIN_PLAYERS = 1;
-    private static final int MAX_PLAYERS = 4;
+    private static final int MIN_OPPONENTS = 1;
+    private static final int MAX_OPPONENTS = 4;
 
     public CommandLineController() {
+        this.gameEngine = new Game(DECK_RESOURCE);
         this.view = new CommandLineView();
         this.scanner = new Scanner(System.in);
         this.logger = new OutputLogger();
     }
 
-    public void startGame() {
-        printWelcomeBanner();
-        newGameHandler();
-    }
-
-    private void newGameHandler(){
-        String input = "";
-        System.out.println("To start a new game, press f\nTo see game statistics, press s");
-        logger.printToLog("Game started");
-        logger.printToLog("New line \n new line too.");
-        input = scanner.nextLine();
+    public void start() {
+        presentWelcomeMessage();
 
+        String input = scanner.nextLine();
+        // TODO: Create an Enum to encapsulate the flags
         if (input.equalsIgnoreCase("f")) {
-            int numberOfPlayers = requestNumberOfPlayers();
-            Game game = new Game(numberOfPlayers, this);
-            System.out.println("Game Start");
-            game.start();
+            // Start the game
+            int numberOfOpponents = requestNumberOfOpponents();
+            startNewGame(numberOfOpponents);
+        } else if (input.equalsIgnoreCase("s")) {
+            // TODO: Start statistics mode
         }
 
-        if (input.equalsIgnoreCase("s")) {
+        scanner.nextLine();
+    }
+
+    private void startNewGame(int numberOfOpponents) {
+        System.out.println("Game Start");
+
+        List<Card> communalPile = new ArrayList<>();
+        int rounderNumber = 0;
```

Comment:
Very minor but should this be round rather than rounder?

Reply:
Hahaha thanks! Sorry, this was a mistake indeed.